crates/ruff/src/rules/flake8_pyi/rules/unnecessary_literal_union.rs

```diff
@@ -0,0 +1,92 @@
+use ruff_python_semantic::SemanticModel;
+use rustpython_parser::ast::{self, Expr, Operator, Ranged};
+
+use ruff_diagnostics::{Diagnostic, Violation};
+use ruff_macros::{derive_message_formats, violation};
+
+use crate::checkers::ast::Checker;
+
+/// ## What it does
+/// Checks for the presence of multiple literal types in a union.
+///
+/// ## Why is this bad?
+/// Literal types accept multiple arguments and it is clearer to specify them as a single literal.
+///
+/// ## Example
+/// ```python
+/// field: Literal[1] | Literal[2]
+/// ```
+///
+/// Use instead:
+/// ```python
+/// field: Literal[1, 2]
+/// ```
+#[violation]
+pub struct UnnecessaryLiteralUnion {
+    members: Vec<String>,
+}
+
+impl Violation for UnnecessaryLiteralUnion {
+    #[derive_message_formats]
+    fn message(&self) -> String {
+        format!(
+            "Multiple literal members in a union. Use a single literal, e.g. `Literal[{}]`",
+            self.members.join(", ")
+        )
+    }
+}
+
+/// PYI030
+pub(crate) fn unnecessary_literal_union(checker: &mut Checker, expr: &Expr) {
+    let mut literal_members = Vec::new();
+    collect_literal_members(&mut literal_members, checker.semantic(), expr);
+
+    // Raise a violation if more than one
+    if literal_members.len() > 1 {
+        let diagnostic = Diagnostic::new(
+            UnnecessaryLiteralUnion {
+                members: literal_members
+                    .into_iter()
+                    .map(|m| checker.locator.slice(m.range()).to_string())
```

Comment:
Nit: consider renaming `m` to `member` here for explicitness? We tend to avoid single-variable names outside of (e.g.) indexes, but it's obviously not a blocker, up to your discretion.